### DISCUSSION TITLE
.NET 9 LINQ Performance Edition

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#105] [General]
# .NET 9 LINQ Performance Edition

As with almost every edition of .NET, the team has been working on improving performance. In this blog post, we will see some improvements to the related tickets and benchmarks.

https://steven-giesel.com/blogPost/783a404a-e39e-480f-bc99-a514a75d752d?utm_source=newsletter.csharpdigest.net&utm_medium=newsletter&utm_campaign=net-9-linq-performance-edition

**Comment by abatishchev** [comment]
Please don't name variables (even if static) in PascalCase, otherwise it's impossible to distinguish instance from static (extension) method calls, e.g. `List.OrderBy()`.

Following the standard naming convention that would be `_list.OrderBy()`.

**Reply by linkdotnet** [reply]
Good point - i will change that later.

by the way, MS itself advised in favor of the given casing: https://learn.microsoft.com/en-us/previous-versions/dotnet/netframework-1.1/x2dbyw72(v=vs.71)?redirectedfrom=MSDN

**Reply by abatishchev** [reply]
That one is very old doc, dating back to .NET 1.1

But so is this one: https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-fields, first published in 2005, so it's .NET 2.0 at best.

Here's a more modern one: https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md

>We use _camelCase for internal and private fields and use readonly where possible. Prefix internal and private instance fields with _, static fields with s_ and thread static fields with t_>

I  _personally_ don't think using an additional prefix
is a good idea, smells like the Hungarian notation to me. Luckily, it's not a very common style across OSS projects. My guess would be that it's coming from the C/C++ nature of .NET Runtime.

And another one: https://google.github.io/styleguide/csharp-style.html#code

> Names of private, protected, internal and protected internal fields and properties: _camelCase.

All the above is basically a matter of taste and personal preferences. So in no way I insist that _mine_ is anyhow better or more correct than _yours_.

The primary reason for my original comment was to clearly distinguish the newly added API (if any) from the rest of the code. Cheers!

**Reply by linkdotnet** [reply]
I am more than keen to change it (as it is really easy). The current style is more or less the default of Rider.

I will change it

**Reply by neon-sunset** [reply]
It's a matter of choice - there's no strict "rule" to abide by (multiple conventions co-exist) for class/struct fields and a different choice can be made for better-looking examples.

**Reply by linkdotnet** [reply]
Oh yeah - but I get the point that it might raise some ambiguity. Therefore I changed it to make it "painfully" clear that there isn't some extension method or whatnot at play.

**Comment by neon-sunset** [comment]
There have been quite a few more improvements for LINQ in .NET 9. For example:

- https://github.com/dotnet/runtime/pull/97905
- https://github.com/dotnet/runtime/pull/96570 (adopts work from this contribution: https://github.com/dotnet/runtime/pull/90459#discussion_r1292257807)

You can generally just filter by label for merged PRs and see there were other impactful changes: https://github.com/dotnet/runtime/pulls?q=is%3Apr+is%3Aclosed+label%3Aarea-System.Linq

**Reply by linkdotnet** [reply]
Good hint and very good approach in general- that is somewhat what I did when I initially skimmed through the repository.
I deemed some of them not as "noteworthy" as they cover some edge cases.
